client/api/omni/specs/omni.proto

```diff
@@ -1251,3 +1251,15 @@ message ClusterMachineRequestStatusSpec {
   string provider_id = 3;
   Stage stage = 4;
 }
+
+// InfraMachineConfigSpec is the spec of the user-owned InfraMachineConfig resource.
+message InfraMachineConfigSpec {
+  enum MachinePowerState {
+    POWER_STATE_DEFAULT = 0;
+    POWER_STATE_OFF = 1;
+    POWER_STATE_ON = 2;
+  }
+
+  MachinePowerState power_state = 1;
+  bool accepted = 2;
+}
```

Comment:
Looks like a duplicate of `InfraMachineStatusSpec`, maybe we can use the same spec for both resources?

Reply:
There are some differences: 
- the zero value in this one (`Config`) is `DEFAULT`, while in the `Status` spec it is `UNKNOWN`, and logically they represent different things.
- in the Config, there's the `accepted` field, in Status, there's `readyToUse` field, again, semantically different.

I can imagine them diverging even more soon, so I'd rather keep them this way for now, if that's not an issue.